realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -724,4 +760,174 @@ public String toString() {
         sb.append("]");
         return sb.toString();
     }
+
+    // Custom RealmList iterator.
+    private class RealmItr implements Iterator<E> {
+        /**
+         * Index of element to be returned by subsequent call to next.
+         */
+        int cursor = 0;
+
+        /**
+         * Index of element returned by most recent call to next or
+         * previous.  Reset to -1 if this element is deleted by a call
+         * to remove.
+         */
+        int lastRet = -1;
+
+        /**
+         * The modCount value that the iterator believes that the backing
+         * List should have.  If this expectation is violated, the iterator
+         * has detected concurrent modification.
+         */
+        int expectedModCount = modCount;
```

Comment:
Do you ever increase the value of `modCount`? Also, will `ConcurrentModificationException` ever be thrown since the realm checking is always ran first?